syntax = "proto3";
import "PersonMessage.proto";

message LocationMessage {
  int32 id = 1;
  int32 person_id = 2;
  string longitude = 3;
  string latitude = 4;
  string creation_time = 5;
}

message ConnectionMessage {
  PersonMessage person = 1;
  LocationMessage location = 2;
}

message ConnectionListMessage {
  repeated ConnectionMessage connections = 1;
}

message ConnectionRequestMessage {
  int32 person_id = 1;
  string start_date = 2;
  string  end_date = 3;
  int32 meters = 4;
}

service ConnectionService {
  rpc FindConnectionList(ConnectionRequestMessage) returns (ConnectionListMessage);
}